src/main/java/org/takes/misc/Condition.java

```diff
@@ -62,4 +72,71 @@ public boolean fits(final T element) {
         }
     }
 
+    /**
+     * Condition that always evaluate to true.
+     * @author Izbassar Tolegen (t.izbassar@gmail.com)
+     * @version $Id$
+     * @param <T> Type of items
+     * @since 2.0
+     */
+    final class True<T> implements Condition<T> {
+
+        @Override
+        public boolean fits(final T element) {
+            return true;
+        }
+    }
+
+    /**
+     * Regexp condition that check items with given regexp.
+     * @author Izbassar Tolegen (t.izbassar@gmail.com)
+     * @version $Id$
+     * @since 2.0
+     */
+    final class Regexp implements Condition<String> {
+
+        /**
+         * Condition.
+         */
+        private final Condition<String> condition;
+
+        /**
+         * Regex to validate against.
+         */
+        private final String regexp;
+
+        /**
+         * Ctor.
+         * @param condition Original condition
+         * @param regexp Regex to validate against
+         */
+        public Regexp(final Condition<String> condition, final String regexp) {
+            this.condition = condition;
+            this.regexp = regexp;
+        }
+
+        /**
+         * Ctor with original condition set to {@link Condition.True}.
+         * @param regexp Regex to validate against
+         */
+        public Regexp(final String regexp) {
+            this(new Condition.True<String>(), regexp);
+        }
+
+        @Override
+        public boolean fits(final String element) {
+            return this.valid(element) && this.condition.fits(element);
+        }
+
+        /**
+         * Checks given element against regex.
+         * @param element Element to check
+         * @return True if element fits regex
+         */
+        private boolean valid(final String element) {
+            final Pattern pattern = Pattern.compile(this.regexp);
```

Comment:
@t-izbassar I think it would be better to turn this variable into private field and assign it in primary constructor: `public Regexp(final Pattern ptn)`

Reply:
@g4s8 this will introduce side effect to constructor with `String` parameter as `Pattern.compile()` can throw exception when regex is invalid. Do we want that?

Reply:
@t-izbassar I think yes, look how it's implemented in `FkRegex`:
```java
    public FkRegex(final String ptn, final Take tke) {
        this(
            Pattern.compile(ptn, Pattern.CASE_INSENSITIVE | Pattern.DOTALL),
            tke
        );
    }
```
Takes user may decide to use string for pattern or `Pattern` object.